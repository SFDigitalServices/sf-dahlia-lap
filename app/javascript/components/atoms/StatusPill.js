import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { LEASE_UP_STATUS_VALUES, getStatusPillClass, getStatusPillLabel } from '../../utils/statusUtils'

const StatusPill = ({ status }) => (
  <div className={classNames('status-pill', getStatusPillClass(status))}>
    {getStatusPillLabel(status)}
  </div>
)

StatusPill.propTypes = {
  status: PropTypes.oneOf(LEASE_UP_STATUS_VALUES)
}

StatusPill.defaultProps = {
  status: null
}

export default StatusPill
