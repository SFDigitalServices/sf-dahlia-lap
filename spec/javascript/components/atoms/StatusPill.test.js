import React from 'react'
import { shallow } from 'enzyme'
import StatusPill from 'components/atoms/StatusPill'

const getWrapper = (status) => shallow(<StatusPill status={status} />)

describe('StatusPill', () => {
  test('should render properly with approved status', () => {
    const wrapper = getWrapper('Approved')
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with withdrawn status', () => {
    const wrapper = getWrapper('Withdrawn')
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Processing status', () => {
    const wrapper = getWrapper('Processing')
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Appealed status', () => {
    const wrapper = getWrapper('Appealed')
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Disqualified status', () => {
    const wrapper = getWrapper('Disqualified')
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Lease Signed status', () => {
    const wrapper = getWrapper('Lease Signed')
    expect(wrapper).toMatchSnapshot()
  })
})
