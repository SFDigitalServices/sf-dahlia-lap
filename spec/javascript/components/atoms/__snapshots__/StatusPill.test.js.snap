// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusPill should render properly with Appealed status 1`] = `
<div
  className="status-pill is-appealed"
>
  Appealed
</div>
`;

exports[`StatusPill should render properly with Disqualified status 1`] = `
<div
  className="status-pill is-disqualified"
>
  Disqualified
</div>
`;

exports[`StatusPill should render properly with Lease Signed status 1`] = `
<div
  className="status-pill is-leased"
>
  Lease Signed
</div>
`;

exports[`StatusPill should render properly with Processing status 1`] = `
<div
  className="status-pill is-processing"
>
  Processing
</div>
`;

exports[`StatusPill should render properly with approved status 1`] = `
<div
  className="status-pill is-approved"
>
  Approved
</div>
`;

exports[`StatusPill should render properly with withdrawn status 1`] = `
<div
  className="status-pill is-withdrawn"
>
  Withdrawn
</div>
`;
